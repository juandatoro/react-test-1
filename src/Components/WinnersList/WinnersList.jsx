import PropTypes from 'prop-types';
import { Title } from 'Components';
import styles from './WinnersList.module.scss';

export const WinnersList = ({ winners = [] }) => {
  return (
    <div className={styles.winners}>
      <Title className={styles.winners__title}>THe winners are</Title>
      {winners.map(({ category, movie }) => (
        <div className={styles.winners__movies} key={category}>
          <p className={styles['winners__text--category']}>{category}</p>
          <p className={styles['winners__text--movie']}>{movie}</p>
        </div>
      ))}
    </div>
  );
};

WinnersList.propTypes = {
  winners: PropTypes.arrayOf(
    PropTypes.shape({
      category: PropTypes.string.isRequired,
      movie: PropTypes.string.isRequired,
    })
  ).isRequired,
};
