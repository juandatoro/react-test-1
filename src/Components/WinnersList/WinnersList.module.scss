.winners {
  color: rgb(var(--dark-shadow));

  &__title {
    margin: 0;
    margin-bottom: var(--spacing-m);
  }
  &__movies {
    &:not(:last-child) {
      margin: 2rem;
    }
  }
  &__text {
    &--category {
      margin-bottom: 1rem;
      font-size: 2.2rem;
      font-weight: bold;
    }
    &--movie {
      font-size: 1.8rem;
    }
  }
}
