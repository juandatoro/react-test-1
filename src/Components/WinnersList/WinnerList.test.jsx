import { render } from '@testing-library/react';
import { WinnersList } from './WinnersList';

describe('WinnersList component', () => {
  const { container } = render(<WinnersList />);

  test('To render main title and submit btn', () => {
    expect(container).toMatchSnapshot();
  });
});
