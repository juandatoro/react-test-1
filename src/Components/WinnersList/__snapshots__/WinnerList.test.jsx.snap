// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnersList component To render main title and submit btn 1`] = `
<div>
  <div
    class="winners"
  >
    <h1
      class="title winners__title"
    >
      THe winners are
    </h1>
  </div>
</div>
`;
